src/core/src/op/util/gather_base.cpp

```diff
@@ -122,6 +122,7 @@ bool evaluate_gather(const ngraph::HostTensorPtr& arg0,
     switch (out->get_element_type()) {
         NGRAPH_TYPE_CASE(evaluate_gather, i32, arg0, arg1, out, axis, batch_dims);
         NGRAPH_TYPE_CASE(evaluate_gather, i64, arg0, arg1, out, axis, batch_dims);
+        NGRAPH_TYPE_CASE(evaluate_gather, i16, arg0, arg1, out, axis, batch_dims);
```

Comment:
why?

Reply:
don't review it now, the PR is not ready and includes different features. I've created it to perform all available tests

Reply:
just wanted to highlight that evaluate must not be extended

Reply:
excluded from PR